### PR TITLE
Fixed link-cycle-detection

### PR DESCRIPTION
**Pull request by ewoutp**
This PR solves the (incorrect) validation error when component A links to B and B links to another service.

### CODE REVIEW

**Comment by ewoutp** [comment]
@ZeissS RFR

File: links_test.go [hunk 1]
```diff
@@ -44,3 +44,45 @@ func TestLinkStringReliability(t *testing.T) {
...
+				Image: MustParseImageDefinition("registry.giantswarm.io/landingpage:0.10.0"),
+				Links: LinkDefinitions{
+					LinkDefinition{
+						Service:    AppName("other-service"),
+						TargetPort: generictypes.MustParseDockerPort("80/tcp"),
+					},
+				},
+				Ports: PortDefinitions{
+					generictypes.MustParseDockerPort("80/tcp"),
+				},
```
Comment (zyndiecate) [inline links_test.go]: I very much liked the way we create test components as seen here: https://github.com/giantswarm/user-config/blob/master/diff_test.go#L121

**Comment by zyndiecate** [comment]
Just a style note. LGTM

File: links_test.go (hunk 1 above)
Comment (ewoutp) [inline links_test.go]: I've tried to be as minimalistics as possible here.

**Pull request merged**